packages/kit/src/runtime/client/client.js

```diff
@@ -1161,6 +1138,32 @@ export function create_client({ target, base, trailing_slash }) {
 			await Promise.all(promises);
 		},
 
+		apply_action: async (result) => {
+			if (result.type === 'error') {
+				const url = new URL(location.href);
+				const intent = get_navigation_intent(url, true);
+				forced_error = result.error;
```

Comment:
this feels racy — probably better if we can pass it as an argument to `update`

Reply:
did a slightly different approach — rather than calling `update`, we skip ahead to creating a new `NavigationState` that includes the error